### PR TITLE
IT-3162: Initial CI

### PR DESCRIPTION
**Pull request by xschildw**
This PR sets up the workflow to deploy to the staging bucket.

### CODE REVIEW

**Review by hallieswan — COMMENTED** [review]
The changes look good to me! A couple of questions --

- @xschildw: just curious, which URL will be used for the staging site? 
- The base branch for this PR is `main`, likely because `main` is the repo's default branch. @syoung-smallwisdom: should we create `staging` and `production` branches and make `staging` the default branch to align the branching strategy between this repo and `open-bridge-web`? Then, we could change the PR's base branch to `staging`, so that the `deploy-open-bridge-static-staging` workflow will run when this PR is merged. What do you think?

**Comment by xschildw** [comment]
It's currently setup as [https://staging.bridgedigital.health/](https://staging.bridgedigital.health/), consistent with [https://staging.studies.bridgedigital.health/](https://staging.studies.bridgedigital.health/).
And I just assumed we'd eventually have the same branches as the the 'open-bridge-web' repo.

File: .github/workflows/deploy-openbridge-static-staging.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+on:
+  push:
+    branches: ["staging"]
+
+jobs:
+  deploy:
+    permissions:
+      id-token: write
+      contents: read
```
Comment (zaro0508) [inline .github/workflows/deploy-openbridge-static-staging.yaml:11]: isn't this already in the `build-and-deploy.yaml` file?

File: .github/workflows/deploy-openbridge-static-staging.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+name: deploy-open-bridge-static-staging
+
+on:
+  push:
+    branches: ["staging"]
```
Comment (zaro0508) [inline .github/workflows/deploy-openbridge-static-staging.yaml:5]: only run CI for staging branch? what about the other branches?

File: .github/workflows/build-and-deploy.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        run: |
+          export CI=false
+          yarn install --check-files
+          yarn build
+      - name: Assume AWS Role
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          aws-region: us-east-1
+          role-to-assume: arn:aws:iam::797640923903:role/sagebase-github-oidc-bridgedigitalhealth-openbridgeweb
+          role-session-name: GitHubActions-${{ github.repository_owner }}-${{ github.event.repository.name }}-${{ github.run_id }}
```
Comment (zaro0508) [inline .github/workflows/build-and-deploy.yaml:47]: might be too long, AWS only allows 64 chars max

**Review by zaro0508 — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy-openbridge-static-staging.yaml (hunk 2 above)
Comment (syoung-smallwisdom) [reply .github/workflows/deploy-openbridge-static-staging.yaml:5]: We also need an action for deploying to production, right? That triggers off of the "production" branch?

Comment (xschildw) [reply .github/workflows/deploy-openbridge-static-staging.yaml:5]: I haven't setup the prod infra yet. Will add later.

File: .github/workflows/build-and-deploy.yaml (hunk 3 above)
Comment (xschildw) [reply .github/workflows/build-and-deploy.yaml:47]: 54, that role already exists (I just added perms and repo to it).

File: .github/workflows/deploy-openbridge-static-staging.yaml (hunk 1 above)
Comment (xschildw) [reply .github/workflows/deploy-openbridge-static-staging.yaml:11]: I thought they needed to be in both (and match). I'll check.

**Review by syoung-smallwisdom — APPROVED** [review]
looks good

File: .github/workflows/deploy-openbridge-static-staging.yaml (hunk 1 above)
Comment (xschildw) [reply .github/workflows/deploy-openbridge-static-staging.yaml:11]: I tried calling a job with the permissions above from a job that did not have them and got 'The nested job 'build' is requesting 'id-token: write', but is only allowed 'id-token: none'.' . Added the perms to the caller and the job runs. Let's get it in to see if it actually deploys...

Comment (zaro0508) [reply .github/workflows/deploy-openbridge-static-staging.yaml:11]: duplication is definitely not needed.  we have a bunch of repos setup where it's only set once.   for example https://github.com/Sage-Bionetworks-IT/organizations-infra and https://github.com/Sage-Bionetworks-Workflows/nextflow-infra

Comment (xschildw) [reply .github/workflows/deploy-openbridge-static-staging.yaml:11]: Wonder if it's because I was in a repo that's outside of the orgs... I'll give a try.

**Pull request merged**